src/java.base/unix/native/libnet/NetworkInterface.c

```diff
@@ -1852,7 +1853,8 @@ static int getMTU(JNIEnv *env, int sock, const char *ifname) {
 static int getFlags(int sock, const char *ifname, int *flags) {
     struct ifreq if2;
     memset((char *)&if2, 0, sizeof(if2));
-    strncpy(if2.ifr_name, ifname, sizeof(if2.ifr_name) - 1);
+    strncpy(if2.ifr_name, ifname, sizeof(if2.ifr_name));
+    if2.ifr_name[sizeof(if2.ifr_name) - 1] = 0;
```

Comment:
Given the use of memset to zero the struct as if it were a pointer to char, this change is unnecessary.  It might reduce warnings, but it creates the potential for merge conflicts (which are the worst part of keeping things up to date).  In general, although we want correct code, we also don't want to make every upstream merge a pain, so over time I've come to the conclusion that we should try to stick with the upstream in cases like this where there isn't a functional change.  You could try submitting the changes upstream first, and then we would get them automatically.

Reply:
I'm aware it has no practical effect, the sole purpose was to get rid of the warning. This also makes the FreeBSD code identical to the code for Linux.

But I agree, it's better to get this updated upstream, possibly with more refactoring to avoid unnecessary duplicated code for the various platforms.